homeassistant/components/vera/config_flow.py

```diff
@@ -0,0 +1,68 @@
+"""Config flow for Vera."""
+import re
+from typing import List
+
+import pyvera as pv
+from requests.exceptions import RequestException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EXCLUDE, CONF_LIGHTS
+
+from .const import CONF_CONTROLLER, DOMAIN
+
+LIST_REGEX = re.compile("[^0-9]+")
+
+
+def parse_int_list(data: str) -> List[int]:
+    """Parse a string into a list of ints."""
+    return [int(s) for s in LIST_REGEX.split(data) if len(s) > 0]
+
+
+class VeraFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Vera config flow."""
+
+    async def async_step_user(self, config: dict = None):
+        """Handle user initiated flow."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason="already_setup")
+
+        if config:
+            new_config = {
+                CONF_CONTROLLER: config.get(CONF_CONTROLLER),
+                CONF_LIGHTS: parse_int_list(config.get(CONF_LIGHTS, "")),
```

Comment:
The main flow should only include required items for the connection. Optional items should go in an options flow.

https://developers.home-assistant.io/docs/en/config_entries_options_flow_handler.html